app/src/main/java/com/android/solvit/provider/ui/request/RequestsFeed.kt

```diff
@@ -54,40 +54,166 @@ import androidx.compose.ui.Alignment
+  val requests = allRequests.filter { it.status == ServiceRequestStatus.PENDING }
+  val selectedRequest = remember { mutableStateOf<ServiceRequest?>(null) }
+  var selectedService by remember { mutableStateOf("All Services") }
+  val searchQuery = remember { mutableStateOf("") }
+  val showDialog = remember { mutableStateOf(false) }
+  val user by authViewModel.user.collectAsState()
+  val providerId = user?.uid ?: "-1"
+  val packages = packageProposalViewModel.proposal.collectAsState()
+  val seekerState = remember { mutableStateOf<SeekerProfile?>(null) }
+  val isReadyToNavigate by chatViewModel.isReadyToNavigate.collectAsState()
+  val isSeekerReady = remember { mutableStateOf(false) }
+  val repliedClicked = remember { mutableStateOf(false) }
+
+  LaunchedEffect(selectedRequest.value) {
+    val seeker = selectedRequest.value?.userId?.let { seekerProfileViewModel.fetchUserById(it) }
+    seekerState.value = seeker
+    isSeekerReady.value = seeker != null // Mark as ready when seeker is fetched
+  }
+
+  LaunchedEffect(isSeekerReady.value) {
+    if (repliedClicked.value && isSeekerReady.value) {
+      selectedRequest.value?.let { request ->
+        chatViewModel.prepareForChat(
+            false, providerId, request.userId, seekerState.value, request.uid)
+      }
+    }
+  }
+
+  LaunchedEffect(isReadyToNavigate) {
+    if (repliedClicked.value && isReadyToNavigate) {
+      navigationActions.navigateTo(Screen.CHAT)
+      chatViewModel.resetIsReadyToNavigate()
+      isSeekerReady.value = false
+      repliedClicked.value = false
+    }
+  }
+
+  Scaffold(
+      topBar = { RequestsTopBar(navigationActions, notificationViewModel, providerId) },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { navigationActions.navigateTo(it.route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION_PROVIDER,
+            selectedItem = navigationActions.currentRoute())
+      }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .background(colorScheme.background)
+                    .testTag("ScreenContent"),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              Spacer(modifier = Modifier.height(8.dp))
+
+              // Search Bar and Service Type Filter
+              Row(
+                  modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp),
+                  horizontalArrangement = Arrangement.spacedBy(8.dp),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    SearchBar(searchQuery, Modifier.weight(1f))
+                    ServiceTypeFilter(
+                        selectedService = selectedService,
+                        onServiceSelected = { selectedService = it },
+                    )
+                  }
+
+              // Filtered Requests List
```

Comment:
Consider moving business logic from RequestsFeedScreen into a ViewModel. This for exemple the request filtering, price validation.

Reply:
same, in case you have time to do it it would be much more elegant to move all this logic

Reply:
Thanks for the advice guys, I'll do just that.